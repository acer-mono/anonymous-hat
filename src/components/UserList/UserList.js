import React from 'react';
import PropTypes from 'prop-types';
import User from '@/components/User';

export default class UserList extends React.Component {
  render() {
    const { list, handleClick } = this.props;
    return (
      <ul className="list-group">
        {list.map(user => (
          <User id={user.id} nickname={user.nickname} key={user.id} handleClick={handleClick} />
        ))}
      </ul>
    );
  }
}

UserList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      nickname: PropTypes.string.isRequired,
    }),
  ),
  handleClick: PropTypes.func.isRequired,
};
