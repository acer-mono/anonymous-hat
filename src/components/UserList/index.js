import UserList from '@/components/UserList/UserList';
export default UserList;
