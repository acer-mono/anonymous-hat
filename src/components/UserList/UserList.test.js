import { shallow } from 'enzyme/build';
import UserList from '@/components/UserList/UserList';
import React from 'react';
import renderer from 'react-test-renderer';

describe('UserList', () => {
  test('add 3 elements', () => {
    const users = [
      { nickname: 'first', id: '1' },
      { nickname: 'second', id: '2' },
      { nickname: 'second', id: '3' },
    ];
    const handler = jest.fn();
    const component = shallow(<UserList list={users} handleClick={handler} />);
    expect(component.find('User')).toHaveLength(3);
  });

  test('add empty array', () => {
    const users = [];
    const handler = jest.fn();
    const component = shallow(<UserList list={users} handleClick={handler} />);
    expect(component.find('User')).toHaveLength(0);
  });

  test('UserList test with snapshot (empty user list)', () => {
    const users = [];
    const handler = jest.fn();
    const component = renderer.create(<UserList list={users} handleClick={handler} />);
    let componentSnapshot = component.toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });

  test('UserList test with snapshot (user list with 3 messages)', () => {
    const users = [
      { nickname: 'first', id: '1' },
      { nickname: 'second', id: '2' },
      { nickname: 'second', id: '3' },
    ];
    const handler = jest.fn();
    const component = renderer.create(<UserList list={users} handleClick={handler} />);
    let componentSnapshot = component.toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });
});
