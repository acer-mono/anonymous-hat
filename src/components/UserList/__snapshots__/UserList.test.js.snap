// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList MessageList test with snapshot (message list with 3 messages) 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      first
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      second
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      second
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`UserList UserList test with snapshot (empty user list) 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
/>
`;

exports[`UserList UserList test with snapshot (user list with 3 messages) 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      first
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      second
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="list-group-item"
    data-rb-event-key={null}
    disabled={false}
    onClick={[Function]}
  >
    <span>
      second
    </span>
    <button
      className="btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
</div>
`;
